Library/Homebrew/livecheck/strategy/sparkle.rb

```diff
@@ -0,0 +1,64 @@
+# typed: false
+# frozen_string_literal: true
+
+require "bundle_version"
+require_relative "page_match"
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      # The {Sparkle} strategy fetches content at a URL and parses
+      # its contents as a Sparkle appcast in XML format.
+      #
+      # @api private
+      class Sparkle
+        extend T::Sig
+
+        NICE_NAME = "Sparkle"
+
+        PRIORITY = 1
+
+        # Whether the strategy can be applied to the provided URL.
+        sig { params(url: String).returns(T::Boolean) }
+        def self.match?(url)
+          url.match?(%r{^https?://}) &&
+            ["application/xml", "text/xml"].include?(Strategy.page_headers(url)["content-type"]) &&
+            Strategy.page_contents(url).include?("http://www.andymatuschak.org/xml-namespaces/sparkle")
+        end
```

Comment:
We should fix the logic to not get to the point where we call this if another strategy with higher priority already matched.

Reply:
We specifically check all the strategies against a URL (instead of ordering strategies by priority and stopping at the first match) so that we can output the list of matching strategies in the debug output and verbose JSON output.

Knowing which strategies apply to a URL is useful when working on `livecheck` blocks (or analyzing the current state of the formulae/casks in a tap) and I don't feel that the technical requirements of one strategy should require us to drop this longstanding feature.

If we don't agree on this, we should solicit feedback from other maintainers who work with livecheck regularly (i.e., stakeholders) to see what they think. When we've faced decisions like this in the past, we've discussed the benefits and detriments and collectively chosen a path forward.